daemon/identity.go

```diff
@@ -18,17 +18,18 @@ import (
 	"github.com/cilium/cilium/api/v1/models"
 	. "github.com/cilium/cilium/api/v1/server/restapi/policy"
 	"github.com/cilium/cilium/pkg/identity"
-	"github.com/cilium/cilium/pkg/identity/cache"
 	"github.com/cilium/cilium/pkg/identity/identitymanager"
 	"github.com/cilium/cilium/pkg/labels"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 
 	"github.com/go-openapi/runtime/middleware"
 )
 
-type getIdentity struct{}
+type getIdentity struct {
+	d *Daemon
+}
 
-func newGetIdentityHandler(d *Daemon) GetIdentityHandler { return &getIdentity{} }
+func newGetIdentityHandler(d *Daemon) GetIdentityHandler { return &getIdentity{d: d} }
```

Comment:
nit: These functions should probably be receiving the field they need rather than the whole daemon..? otherwise why not just define these functions with a daemon receiver?

Reply:
This is the convention followed by the rest of the daemon package. I actually like this more, because it decouples our API from the daemon implementation itself. I think the daemon should be its own package, separate from main, eventually ;) .

Reply:
I was meaning something more like:

`func newGetIdentityHandler(identityAllocator *CachingIdentityAllocator) GetIdentityHandler ...`

This way we really are starting to decouple the API from the daemon, rather than just receiving the daemon or passing it as an argument.

Reply:
Will have the `CachingIdentityAllocator` stored in the wrapper type instead of the `Daemon`.